src/transformers/models/xglm/modeling_xglm.py

```diff
@@ -490,7 +490,6 @@ class XGLMPreTrainedModel(PreTrainedModel):
     config_class = XGLMConfig
     base_model_prefix = "model"
     supports_gradient_checkpointing = True
-    _no_split_modules = ["XGLMDecoderLayer"]
```

Comment:
the embedding code is directly inside `XGLMModel`, and if this is in _no_split_modules, basically the whole model is not split

Reply:
Same as for OPT, the layer still needs to stay.